src/components/ConfirmModal.js

```diff
@@ -0,0 +1,40 @@
+import Modal from "./Modal.js";
+
+export default class ConfirmModal extends Modal {
+  constructor(modalSelector) {
+    super({ modalSelector });
+    this._modalForm = this._modalElement.querySelector(".modal__form");
+    this._confirmButton = this._modalElement.querySelector(".modal__button");
+
+    this._submitForm = this._submitForm.bind(this);
+  }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  renderLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._confirmButton.textContent = "Deleting...";
+    } else {
+      this._confirmButton.textContent = submitSave;
+    }
+  }
+
+  _submitForm() {
+    this._handleFormSubmit();
+  }
+
+  close() {
+    this._modalForm.removeEventListener("submit", this._submitForm);
+    super.close();
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._modalForm.addEventListener("submit", (evt) => {
```

Comment:
Needs correction: The listener should be added by reference, so you will be able to remove it on line 29